dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/notification/InMemoryNotifier.java

```diff
@@ -72,6 +73,11 @@ public Notifier notify( JobConfiguration id, NotificationLevel level, String mes
         {
             Notification notification = new Notification( level, id.getJobType(), new Date(), message, completed );
 
+            if ( id.isInMemoryJob() && !StringUtils.isEmpty( id.getUid() ) )
+            {
+                notification.setUid( id.getUid() );
```

Comment:
@mortenoh, I came across this while looking into a Redis bug. Can you remember why we might want to set the `Notification` `UID` to be the same as the Job UID?
This only occurs for the `InMemoryNotifier`. The `RedisNotifier` `Notification`s all have their own unique UIDs.